sheets/utils.py

```diff
@@ -403,17 +404,26 @@ def get_enumerated_data_rows(
         yield from enumerated_data_rows
 
 
-def spreadsheet_repr(spreadsheet):
+def spreadsheet_repr(spreadsheet=None, spreadsheet_metadata=None):
     """
     Returns a simple string representation of a Spreadsheet object
 
     Args:
-        spreadsheet (pygsheets.spreadsheet.Spreadsheet):
+        spreadsheet (pygsheets.spreadsheet.Spreadsheet or None):
+        spreadsheet_metadata (dict or None): A dict of spreadsheet metadata
 
     Returns:
         str: String representation of the spreadsheet
     """
-    return "'{}', id: {}".format(spreadsheet.title, spreadsheet.id)
+    if spreadsheet:
+        sheet_id, title = spreadsheet.id, spreadsheet.title
+    elif spreadsheet_metadata:
+        sheet_id, title = spreadsheet_metadata["id"], spreadsheet_metadata["name"]
+    else:
+        sheet_id, title = None, None
+    if not sheet_id or not title:
```

Comment:
Is it legal for `title` to be an empty string?

Reply:
This is just here to handle cases where neither `spreadsheet` or `spreadsheet_metadata` is passed in, or if a metadata dict with the wrong keys is passed in